### PR TITLE
{Network} `az network vnet-gateway`: Migrate subcommands

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/commands.py [hunk 1]
```diff
@@ -833,29 +832,29 @@ def _make_singular(value):
...
+        self.command_table['network vnet-gateway revoked-cert create'] = VnetGatewayRevokedCertCreate(loader=self)
+
+    with self.command_group('network vnet-gateway root-cert'):
+        from .custom import VnetGatewayRootCertCreate
+        self.command_table['network vnet-gateway root-cert create'] = VnetGatewayRootCertCreate(loader=self)
+
+    with self.command_group('network vnet-gateway ipsec-policy') as g:
+        from .custom import VnetGatewayIpsecPolicyAdd
+        self.command_table['network vnet-gateway ipsec-policy add'] = VnetGatewayIpsecPolicyAdd(loader=self)
+        g.custom_command('clear', 'clear_vnet_gateway_ipsec_policies')
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/commands.py:846]: Should `az network vnet-gateway ipsec-policy clear` support --no-wait?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -7489,53 +7476,23 @@ def show_vpn_connection_device_config_script(cmd, client, resource_group_name, v
...
 
+        return args_schema
 
-def clear_vnet_gateway_ipsec_policies(cmd, resource_group_name, gateway_name, no_wait=False):
-    ncf = network_client_factory(cmd.cli_ctx).virtual_network_gateways
-    gateway = ncf.get(resource_group_name, gateway_name)
-    try:
-        gateway.vpn_client_configuration.vpn_client_ipsec_policies = None
-    except AttributeError:
-        raise CLIError('VPN client configuration must first be set through `az network vnet-gateway create/update`.')
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:7524]: Such exception handling seems missed.

**Comment by necusjz** [comment]
`az network vnet-gateway nat-rule remove` has the similar issues as `az network vnet-gateway ipsec-policy clear`.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -7486,12 +7486,16 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
-def clear_vnet_gateway_ipsec_policies(cmd, resource_group_name, gateway_name):
-    args = {"resource_group": resource_group_name,
-            "name": gateway_name,
-            "vpn_client_ipsec_policies": None}
+def clear_vnet_gateway_ipsec_policies(cmd, resource_group_name, gateway_name, no_wait=False):
+    class VnetGatewayIpsecPoliciesClear(_VnetGatewayUpdate):
+        def pre_operations(self):
+            args = self.ctx.args
+            args.no_wait = no_wait
+
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:7494]: `gateway.vpn_client_configuration.vpn_client_ipsec_policies = None` still missing. As well as other similar logics within `try` block.

**Review by necusjz — APPROVED** [review]

**Pull request merged**